pkg/mcs/scheduling/server/grpc_service.go

```diff
@@ -298,47 +299,39 @@ func (s *Service) AskBatchSplit(_ context.Context, request *schedulingpb.AskBatc
 	// PD doesn't support allocating IDs in batch. We need to allocate IDs
 	// for each region.
 	if requestIDCount != count {
-		// use old way to split region
-		for range splitCount {
-			newRegionID, _, err := c.AllocID(1)
+		for range requestIDCount - 1 {
+			id, _, err = c.AllocID(1)
```

Comment:
Can we guarantee that the IDs returned from consecutive requests are sequential? What if another client makes an ID request in the meantime?

Reply:
Maybe we can save these ids and handle them together later.

Reply:
I plan to revert to the original implementation. /cc @bufferflies

Reply:
I have added a concurrency test.